Arrowgene.Ddon.GameServer/Handler/PawnGetNoraPawnListHandler.cs

```diff
@@ -17,9 +23,14 @@ public PawnGetNoraPawnListHandler(DdonGameServer server) : base(server)
 
         public override PacketId Id => PacketId.C2S_PAWN_GET_NORA_PAWN_LIST_REQ;
 
-        public override void Handle(GameClient client, IPacket packet)
+        public override S2CPawnGetNoraPawnListRes Handle(GameClient client, C2SPawnGetNoraPawnListReq request)
         {
-            client.Send(GameFull.Dump_118);
+            // client.Send(GameFull.Dump_118);
+            S2CPawnGetNoraPawnListRes result = new S2CPawnGetNoraPawnListRes.Serializer().Read(GameFull.Dump_118.AsBuffer());
+
+            List<uint> pawnIds = Server.Database.SelectAllPlayerPawns(uint.MaxValue);
+            result.NoraPawnList = pawnIds.OrderBy(x => Random.Shared.Next()).Take(100).Select(x => new CDataRegisterdPawnList() { PawnId = x }).ToList();
```

Comment:
> Two additional thoughts:
> 
> 1. We should probably trim the client's own pawns from this list.
> 2. `OrderBy` is quicksort and thus O(N logN) ~ O(N^2) in complexity. If `pawnIds` is significantly larger than 100, Fischer Yates might be a better choice here.

This is so small, the time will barely be noticeable and I think unless we see real world impact, it is not worth the time to optimize. Additionally, unless these other sorting algorithms are part of the std library, it is probably not worth the complexity to roll our own.

Reply:
> Also, if INSTANCE_ENCOUNTER_PAWN_NTC works the way I think it does, we need to provide the full data structure, not just the IDs, because that notice is expecting at least the levels to be filled in.

The captured data from the pcap is an empty structure, so in this handler, we do not need the full data. If we need it for other use case, it is available for population via the static function `GameStructure.CDataPawnInfo()`. Alternativly we could move this logic into the pawn class itself, but the legacy code uses this static function instead.